BUILDING.md

```diff
@@ -59,6 +59,12 @@ note1 - The gcc4.8-libs package needs to be installed, because node
   by Joyent. SmartOS images >= 16.4 are not supported because
   GCC 4.8 runtime libraries are not available in their pkgsrc repository
 
+*Note*: On Windows, running Node.js in windows terminal emulators like `mintty`
+  requires the usage of [winpty](https://github.com/rprichard/winpty) for
+  Node's tty channels to work correctly (e.g. `winpty node.exe script.js`).
+  In "Git bash" if you call the node shell wrapper (`node` without the `.exe`
+  extension), `winpty` is used automatically.
```

Comment:
I'd avoid adding this note because this is an implementation detail of "Git Bash" that could change anytime.

Reply:
But it's also a really odd thing that isn't documented anywhere else, and the difference between `node` and `node.exe` is very unobvious.

I think it's worth documenting that if you're on Git Bash, you should just use `node` without the `.exe` to get the required behaviour, so far I've only seen it documented in Github comments (including by you 😁 ) , e.g. https://github.com/nodejs/node/issues/5620#issuecomment-253974666 and https://github.com/nodejs/node/issues/3006#issuecomment-266663186.

>this is an implementation detail of "Git Bash" that could change anytime.

🤷‍♂️ , if it changes we can change the docs, but I wouldn't call it an implementation detail, this is an alias `Git Bash` added to make life easier for Node users.

Reply:
I'm ambivalent. I had both arguments in my head ("GfW quirk" vs. "Undocumented helper")...

Reply:
so, what's the final decision here? any changes needed from my side?

I personally think that it'd be better to mention it now and change later if needed, for all it's worth.